hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3Utils.java

```diff
@@ -63,64 +64,44 @@ private S3Utils() {
+      String s3StorageTypeHeader, String s3StorageConfigHeader,
+      ReplicationConfig clientConfiguredReplConfig,
       ReplicationConfig bucketReplConfig)
       throws OS3Exception {
-    ReplicationConfig clientDeterminedReplConfig = null;
 
-    // Let's map the user provided s3 storage type header to ozone s3 storage
-    // type.
-    S3StorageType s3StorageType = null;
-    if (s3StorageTypeHeader != null && !s3StorageTypeHeader.equals("")) {
-      s3StorageType = toS3StorageType(s3StorageTypeHeader);
+    // If user provided s3 storage type header is not null then map it
+    // to ozone replication config
+    if (!StringUtils.isEmpty(s3StorageTypeHeader)) {
+      return toReplicationConfig(s3StorageTypeHeader, s3StorageConfigHeader);
     }
 
-    boolean isECBucket = bucketReplConfig != null && bucketReplConfig
-        .getReplicationType() == HddsProtos.ReplicationType.EC;
-
-    // if bucket replication config configured with EC, we will give high
-    // preference to server side bucket defaults.
-    // Why we give high preference to EC is, there is no way for file system
-    // interfaces to pass EC replication. So, if one configures EC at bucket,
-    // we consider EC to take preference. in short, keys created from file
-    // system under EC bucket will always be EC'd.
-    if (isECBucket) {
-      // if bucket is EC, don't bother client provided configs, let's pass
-      // bucket config.
-      clientDeterminedReplConfig = bucketReplConfig;
-    } else {
-      // Let's validate the client side available replication configs.
-      boolean isUserPassedReplicationInSupportedList =
-          s3StorageType != null && (s3StorageType.getFactor()
-              .getValue() == ReplicationFactor.ONE.getValue() || s3StorageType
-              .getFactor().getValue() == ReplicationFactor.THREE.getValue());
-      if (isUserPassedReplicationInSupportedList) {
-        clientDeterminedReplConfig = ReplicationConfig.fromProtoTypeAndFactor(
-            ReplicationType.toProto(s3StorageType.getType()),
-            ReplicationFactor.toProto(s3StorageType.getFactor()));
-      } else {
-        // API passed replication number is not in supported replication list.
-        // So, let's use whatever available in client side configured.
-        // By default it will be null, so server will use server defaults.
-        clientDeterminedReplConfig = clientConfiguredReplConfig;
-      }
-    }
-    return clientDeterminedReplConfig;
+    // If client configured replication config is null then default to bucket replication
+    // otherwise default to server side default replication config.
+    return (clientConfiguredReplConfig != null) ?
+        clientConfiguredReplConfig : bucketReplConfig;
   }
 
-  public static S3StorageType toS3StorageType(String storageType)
+  public static ReplicationConfig toReplicationConfig(String storageType, String storageConfig)
```

Comment:
no test

Reply:
All tests for `toReplicationConfig()` are covered by testing `resolveS3ClientSideReplicationConfig` in `TestS3Utils.java`